code/obj/item/pens_writing_etc.dm

```diff
@@ -379,14 +379,22 @@
 			else
 				src.font_color = random_saturated_hex_color(1)
 				src.color_name = hex2color_name(src.font_color)
+				src.color = src.font_color
 
 		write_on_turf(var/turf/T as turf, var/mob/user as mob, params)
 			if (!T || !user || src.in_use || get_dist(T, user) > 1)
 				return
 			src.font_color = random_saturated_hex_color(1)
 			src.color_name = hex2color_name(src.font_color)
+			src.color = src.font_color
 			..()
 
+		afterattack(obj/target, mob/user)
+			if (istype(target, /obj/decal/poster/banner)) // so they alter color when used on a banner
```

Comment:
why are you doing afterattack and not attackby on the actual banner object - this block of code does not make sense to me